### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by NazarLysyi**

### CODE REVIEW

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java [hunk 1]
```diff
@@ -24,18 +24,41 @@
...
+            double root = -c / b;
+            return "Количество решений: 1. Корень: " + formatRoot(root);
+        }
+
+        double discriminantValue = b * b - 4 * a * c;
+
+        if (discriminantValue > 0) {
+            double root1 = (-b - Math.sqrt(discriminantValue)) / (2 * a);
+            double root2 = (-b + Math.sqrt(discriminantValue)) / (2 * a);
+            return "Количество решений: 2. Корни: " + formatRoot(Math.min(root1, root2)) + ";" + formatRoot(Math.max(root1, root2));
```
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task4/Task4.java:48]: Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter1/task4/Task4.java:48: Line is longer than 120 characters (found 132). [LineLength]

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java [hunk 2]
```diff
@@ -24,18 +24,41 @@
...
+        } else if (discriminantValue == 0) {
+            double root = -b / (2 * a);
+            return "Количество решений: 1. Корень: " + formatRoot(root);
+        } else {
+            return "Количество решений: 0.";
+        }
+    }
 
-        return null; // Заглушка. При реализации - удалить
+    static String formatRoot(double root) {
```
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task4/Task4.java:57]: Зачем ЄТО?

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java [hunk 3]
```diff
@@ -24,18 +24,41 @@
...
+        double c = 2;
         System.out.println(solveEquation(a, b, c));
-
     }
 
     static String solveEquation(double a, double b, double c) {
-        //        Место для вашего кода
+        if (a == 0 && b == 0 && c == 0) {
+            return "Бесконечное множество решений.";
+        } else if (a == 0 && b == 0) {
```
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task4/Task4.java:36]: Если в блоке if есть return, то else не пишем. продолжаем как отдельніій if если бііло if/else или просто пишем дальше код если просто if.

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java (hunk 1 above)
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task4/Task4.java:48]: И тестіі тоже пройди =) 
> Task :test FAILED

Task4Test > solveEquationTest(String, double, double, double) > [15] Количество решений: 1. Корень: 1.0, 0, -2, 2 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [16] Количество решений: 1. Корень: 2.0, 0, -2, 4 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [17] Количество решений: 1. Корень: 4.0, 0, -2, 8 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [18] Количество решений: 1. Корень: 2.0, 0, 20, -[40](https://github.com/KFalcon2022/intensive-tasks-2024/actions/runs/11924960191/job/33238308039?pr=209#step:6:41) FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [19] Количество решений: 1. Корень: 8.0, 0, 10, -80 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [20] Количество решений: 1. Корень: 0.0, 1, 0, 0 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [21] Количество решений: 1. Корень: 1.0, 1, -2, 1 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [22] Количество решений: 1. Корень: 2.0, 1, -4, 4 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [23] Количество решений: 1. Корень: 3.0, 1, -6, 9 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [24] Количество решений: 1. Корень: 4.0, 1, -8, 16 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [25] Количество решений: 1. Корень: 5.0, 1, -10, 25 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [26] Количество решений: 2. Корни: -5.0;-4.0, 1, 9, 20 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [27] Количество решений: 2. Корни: -3.0;-2.0, 1, 5, 6 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18


30 tests completed, 16 failed
Task4Test > solveEquationTest(String, double, double, double) > [28] Количество решений: 2. Корни: 3.0;5.0, 1, -8, 15 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [29] Количество решений: 2. Корни: -9.0;5.0, 1, 4, -[45](https://github.com/KFalcon2022/intensive-tasks-2024/actions/runs/11924960191/job/33238308039?pr=209#step:6:46) FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

Task4Test > solveEquationTest(String, double, double, double) > [30] Количество решений: 2. Корни: -1.0;4.0, -1, 3, 4 FAILED
    org.opentest4j.AssertionFailedError at Task4Test.java:18

FAILURE: Build failed with an exception.

**Review by Zawadskiy — CHANGES_REQUESTED** [review]

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java (hunk 2 above)
Comment (NazarLysyi) [reply src/main/java/com/walking/intensive/chapter1/task4/Task4.java:57]: в тестах была ошибка с 1.0, сделал форматирование вывода. не знал что есть правки

File: src/main/java/com/walking/intensive/chapter1/task4/Task4.java (hunk 1 above)
Comment (NazarLysyi) [reply src/main/java/com/walking/intensive/chapter1/task4/Task4.java:48]: код написан до правок в тестах) ребейс сделал, ща поправлю

**Pull request closed without merging**